pyoxidizer/src/distutils/unixccompiler.py

```diff
@@ -251,12 +249,14 @@ def extension_link_shared_object(self,
 
         # Write out a file with the information about the extension. PyOxidizer
         # will read this to know how to ingest the extension.
-        json_path = os.path.join(dest_path, 'extension.%s.json' % name)
-        with open(json_path, 'w', encoding='utf-8') as fh:
+        with open(get_extension_json_path(name), 'w', encoding='utf-8') as fh:
             data = {
+                'dist_name': self.dist.get_name(),
+                'dist_version': self.dist.get_version(),
                 'name': '%s.%s' % (package, name) if package else name,
                 'objects': object_paths,
                 'output_filename': our_output_filename,
+                'restore_filename': output_filename,
                 'libraries': libraries or [],
                 'library_dirs': library_dirs or [],
                 'runtime_library_dirs': runtime_library_dirs or [],
```

Comment:
Compiler name and version, compiler flags. Also package extras, as in `package[extra]`. I think that covers it. Maybe also minimum required glibc version (I'm not sure if that's determined during compile or during link).

Reply:
compiler name and version are not easily obtainable at this stage; `unixccompiler.py` just uses `cc` and then internally when it needs to invoke it performs `PATH` search. 
Do you want me to do the same?

Reply:
We're going to have to remember this I guess.

Reply:
Is there an easy way to clear the cache of CircleCI?